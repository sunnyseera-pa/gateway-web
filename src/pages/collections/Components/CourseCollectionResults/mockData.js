export const data = [
	{
		_id: '5f96f3450d236c6cafdaa473',
		domains: [],
		award: ['BSc'],
		courseOptions: [
			{
				flexibleDates: true,
				_id: '5f96f3450d236c6cafdaa474',
				startDate: null,
				studyMode: 'Full-time',
				studyDurationNumber: 1,
				studyDurationMeasure: 'Month',
				fees: [
					{
						_id: '5f96f3450d236c6cafdaa475',
						feeDescription: 'fefwe',
						feeAmount: 10,
						feePer: 'Week',
					},
					{
						_id: '5f96fcec8f67f67c62cb5b4a',
						feeDescription: 'fewf',
						feeAmount: 122,
						feePer: 'Year',
					},
				],
			},
			{
				flexibleDates: false,
				_id: '5f96f3450d236c6cafdaa476',
				startDate: '2020-11-12T00:00:00.000Z',
				studyMode: 'Full-time',
				studyDurationNumber: 5,
				studyDurationMeasure: 'Day(s)',
				fees: [
					{
						_id: '5f96f3450d236c6cafdaa477',
						feeDescription: 'Test',
						feeAmount: 4,
						feePer: 'Month',
					},
				],
			},
		],
		id: 3491286126459112,
		type: 'course',
		activeflag: 'active',
		title: 'Test A course',
		provider: 'Test A',
		description: 'Test A Test A Test A edited Test A',
		counter: 213,
		relatedresources: 3,
		updated: 1621465200000,
	},
];

export const relatedObjects = [
	{
		_id: '60a61f96ba73364b8606d007',
		objectId: '3491286126459112',
		reason: '',
		objectType: 'course',
		pid: '',
		user: 'Lewis Kearsey',
		updated: '20 May 2021',
	},
];
