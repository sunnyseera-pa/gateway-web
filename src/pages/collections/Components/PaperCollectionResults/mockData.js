export const data = [
	{
		_id: '5fac1d0e87a75fac60037cf3',
		categories: {
			programmingLanguage: [],
		},
		tags: {
			features: [],
			topics: [],
		},
		authors: [5999816745288831],
		id: 59596737286982110,
		type: 'paper',
		name: 'test ppaper email',
		description: 'test ppaper email',
		activeflag: 'active',
		programmingLanguage: [],
		counter: 4,
		relatedresources: 0,
		persons: [
			{
				_id: '6013e9961edb6e93b2814701',
				id: 5999816745288831,
				firstname: 'Rich',
				lastname: 'Hobbs',
			},
		],
		updated: 1621465200000,
	},
];

export const relatedObjects = [
	{
		_id: '60a61f96ba73364b8606cff1',
		objectId: '59596737286982110',
		reason: '',
		objectType: 'paper',
		pid: '',
		user: 'Lewis Kearsey',
		updated: '20 May 2021',
	},
];
