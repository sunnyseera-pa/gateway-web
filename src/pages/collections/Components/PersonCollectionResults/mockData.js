export const data = [
	{
		_id: '5e544facbd427b6e9cd9059e',
		id: 900000011,
		lastname: 'Nei',
		bio: 'Jefferson Comprehensive Epilepsy Center',
		firstname: 'Maromi',
		type: 'person',
		tags: ['Neuroendocrinology', 'Epilepsy', 'Neuroscience'],
		activeflag: 'active',
		authors: [''],
		counter: 75,
		relatedresources: 0,
		persons: [],
		updated: 1621551600000,
	},
];

export const relatedObjects = [
	{
		_id: '60a7bba5bfd44334afc2d069',
		objectId: '900000011',
		reason: '',
		objectType: 'person',
		pid: '',
		user: 'Ciara Ward',
		updated: '21 May 2021',
	},
];
