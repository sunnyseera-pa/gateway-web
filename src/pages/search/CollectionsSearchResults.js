const CollectionsSearchResults = props => {
	const mapResults = React.useCallback(data => {
		return data.map(collection => {
			return (
				<Col sm={12} md={12} lg={6} style={{ 'text-align': '-webkit-center' }}>
					<CollectionCard key={collection.id} data={collection} />
				</Col>
			);
		});
	}, []);

	return <SearchResults {...props} results={mapResults} />;
};
