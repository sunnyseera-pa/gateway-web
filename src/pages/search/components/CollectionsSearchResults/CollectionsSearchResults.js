import React from 'react';
import { Col, Row } from 'react-bootstrap';
import { omit } from '../../../../configs/propTypes';
import CollectionCard from '../../../commonComponents/collectionCard/CollectionCard';
import SearchResults from '../../../commonComponents/SearchResults';
import { PROP_TYPES_SEARCH_RESULTS } from '../../../commonComponents/SearchResults/SearchResults.propTypes';

const CollectionsSearchResults = props => {
	const mapResults = React.useCallback(data => {
		return (
			<Row className='mt-2'>
				{data.map(collection => {
					return (
						<Col sm={12} md={12} lg={6} style={{ 'text-align': '-webkit-center' }}>
							<CollectionCard key={collection.id} data={collection} />
						</Col>
					);
				})}
			</Row>
		);
	}, []);

	return <SearchResults type='collection' results={mapResults} {...props} />;
};

CollectionsSearchResults.propTypes = omit(PROP_TYPES_SEARCH_RESULTS, ['type', 'results']);

export default CollectionsSearchResults;
