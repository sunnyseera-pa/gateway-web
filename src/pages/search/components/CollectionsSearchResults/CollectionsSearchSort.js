import React from 'react';
import SortDropdown from '../SortDropdown';

const CollectionsSearchSort = ({ sort, onSort, search, ...outerProps }) => {
	return (
		<SortDropdown
			handleSort={onSort}
			sort={sort === '' ? (search === '' ? 'latest' : 'relevance') : sort}
			dropdownItems={['relevance', 'popularity', 'latest', 'resources']}
			{...outerProps}
		/>
	);
};

export default CollectionsSearchSort;
