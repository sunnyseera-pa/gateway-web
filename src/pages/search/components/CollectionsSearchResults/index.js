import CollectionsSearchResults from './CollectionsSearchResults';

export default CollectionsSearchResults;
