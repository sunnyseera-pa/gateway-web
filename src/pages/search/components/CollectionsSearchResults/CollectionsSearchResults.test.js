import { render } from '@testing-library/react';
import React from 'react';
import CollectionsSearchResults from '.';
import { data } from './mockData';

const mockCollectionCard = jest.fn();

jest.mock('../../../commonComponents/collectionCard/CollectionCard', () => props => {
	mockCollectionCard(props);
	return <div />;
});

let wrapper;

const props = {
	data,
	count: 656,
	pageNumber: 3,
	totalPages: 10,
	type: 'collections',
	search: 'search term',
	updateOnFilterBadge: jest.fn(),
	onPagination: jest.fn(),
	maxResults: 40,
	isLoading: false,
	sort: 'Sort goes here...',
};

describe('Given the CollectionsSearchResults component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<CollectionsSearchResults {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then has the correct number of results', () => {
			expect(mockCollectionCard).toHaveBeenCalledTimes(40);
		});

		it('Then passes the correct props', () => {
			expect(mockCollectionCard.mock.calls[0][0]).toEqual({
				data: data[0],
			});
		});
	});
});
