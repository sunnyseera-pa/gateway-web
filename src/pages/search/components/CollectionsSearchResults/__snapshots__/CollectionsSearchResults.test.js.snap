// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CollectionsSearchResults component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="text-right save-dropdown"
    >
      Sort goes here...
    </div>
  </div>
  <div
    class="mt-2 row"
  >
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
    <div
      class="col-lg-6 col-md-12 col-sm-12"
      style="text-align: -webkit-center;"
    >
      <div />
    </div>
  </div>
  <ul
    class="pagination"
  >
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        3
      </a>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        4
        <span
          class="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        5
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        6
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        7
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        8
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        9
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        10
      </a>
    </li>
  </ul>
</div>
`;
